### PR TITLE
Polishing for release 4

### PR DESCRIPTION
**Pull request by adiherzog**
@bruderol I did a few cleanup-tasks that I had on my list for a while. I also reviewed the changelog and migration guide. Looks very good!

It's probably easiest to review this PR commit by commit.

### CODE REVIEW

**Comment by bruderol** [comment]
@adiherzog Why do you write: dont delete this branch?? 
If it is important that you want to keep it ---> go to settings and protect the branch.
Otherwise a deleted branch can also be restored again.

**Comment by adiherzog** [comment]
@bruderol Actually we could also delete it and just merge the release branch back to develop.

**Comment by bruderol** [comment]
@adiherzog yes, exactly, we will do that anyway, see my other PR

**Comment by bruderol** [comment]
@adiherzog I will try to review it soon. Thanks!

File: docs/features/Scenarioo-Writer-Documentation-Format.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ A scenario is made up of steps. Each step describes one interaction event in you
 * For each step, there is an XML file in the steps folder (complex type `step` in XML schema), this step files are just numbered like 000.xml, 001.xml, etc.
 * Each step references an image file inside the 'screenshots' directory through the mandatory field `stepDescription.screenshotFileName`. This screenshots can be of any web image format: we recommend PNG, but could also be JPEG, whichever better fits your application or the current step, depending on the content of the current screen. Usually PNG is the best fit, unless a screen is graphically very complex and contains a lot of bitmap images like fotos.
 * Mandatory fields: 
-  * `stepDescription` (complexType): contains the most important properties to describe a step (also displayed on the scenarioo overview for each step of the scenarioo):
+  * `stepDescription` (complexType): contains the most important properties to describe a step (also displayed on the Scenarioo overview for each step of the scenarioo):
```
Comment (bruderol) [inline docs/features/Scenarioo-Writer-Documentation-Format.md]: last word is wrong: should be "scenario" (without two os)

File: docs/setup/Migration-Guide.md [hunk 2]
```diff
@@ -4,59 +4,62 @@ Guide on how to upgrade for major versions of scenarioo.
...
-           * you can later remove those settings for the config file location from your context.xml, if you configured it there, since the location is now fix in your data directory and can not be configured separately anymore in newer versions. 
-        * If you do not find the config file at all: this might mean that you never saved your configuration, scenarioo might just be running using its internal default configuration. Go to `Manage`-Tab in the scenarioo viewer web app and choose `General Settings / Save` this will save the file for you with current settings.
-     * When you found the config file then copy it to the scenarioo data directory
-        * Copy the file to the root of your scenarioo data path (see step 0 if you forgot)
+           * you will find the `config.xml` file under `{user.home}/.scenarioo` or in the config directory that you configured in your webservers `context.xml` file in property `scenariooConfigurationDirectory` or `configurationDirectory`.
```
Comment (bruderol) [inline docs/setup/Migration-Guide.md]: not agree "webserver's" was right here, this is not a plural!

File: docs/setup/Migration-Guide.md [hunk 3]
```diff
@@ -4,59 +4,62 @@ Guide on how to upgrade for major versions of scenarioo.
...
+           * the file might have been even configured to have a different name than `config.xml` in your `context.xml` file using property `scenariooConfigurationFilename` or `configurationFilename` (you will have to rename that file to `config.xml` after you copied it!).
```
Comment (bruderol) [inline docs/setup/Migration-Guide.md]: oh, ok well spotted - thanks!

File: scenarioo-client/gulpfile.js [hunk 4]
```diff
@@ -136,8 +136,8 @@ gulp.task('test-watch', ['environmentConstants'], function (done) {
 /**
  * Run protractor ui tests (without generating Scenarioo documentation)
  */
-gulp.task('test-e2e', function () {
-    gulp.src(['./test/protractorE2E/specs/**/*.js'])
+gulp.task('e2e', function () {
+    gulp.src(['./test/e2e/specs/**/*.js'])
```
Comment (bruderol) [inline scenarioo-client/gulpfile.js]: OH THANKS GGOD!! I like that you cleaned this finally up! 👍 👍 
`protractorE2E` was a real bad directory name, very unusal.

File: scenarioo-client/package.json [hunk 5]
```diff
@@ -16,15 +16,15 @@
...
-        "test-e2e-before": "npm run lint-ts-e2e && webdriver-manager update",
-        "test-e2e": "npm run test-e2e-before && protractor protractor-e2e.conf.js ",
-        "test-e2e-scenarioo": "npm run test-e2e-before && protractor protractor-e2e-scenarioo.conf.js",
+        "e2e-before": "npm run lint-ts-e2e && webdriver-manager update",
+        "e2e": "npm run e2e-before && protractor protractor-e2e.conf.js ",
+        "e2e-scenarioo": "npm run e2e-before && protractor protractor-e2e-scenarioo.conf.js",
         "webpack-dev-server": "webpack-dev-server",
         "start": "webpack-dev-server -d --inline --progress --port 8500",
         "start-dev": "webpack-dev-server -d --inline --progress --port 8500 --watch",
         "lint-all": "npm run lint-js && npm run lint-ts-e2e",
```
Comment (bruderol) [inline scenarioo-client/package.json]: I think "lint-all" should simply be "lint" as usual

File: scenarioo-client/package.json [hunk 6]
```diff
@@ -16,15 +16,15 @@
         "test": "npm run lint-all && karma start --single-run",
         "test-watch": "karma start --auto-watch --no-single-run",
         "webdriver-manager": "webdriver-manager",
-        "test-e2e-before": "npm run lint-ts-e2e && webdriver-manager update",
-        "test-e2e": "npm run test-e2e-before && protractor protractor-e2e.conf.js ",
-        "test-e2e-scenarioo": "npm run test-e2e-before && protractor protractor-e2e-scenarioo.conf.js",
+        "e2e-before": "npm run lint-ts-e2e && webdriver-manager update",
+        "e2e": "npm run e2e-before && protractor protractor-e2e.conf.js ",
+        "e2e-scenarioo": "npm run e2e-before && protractor protractor-e2e-scenarioo.conf.js",
```
Comment (bruderol) [inline scenarioo-client/package.json]: I like it

**Comment by adiherzog** [comment]
@bruderol Thanks for the feedback. I implemented the three changes and will now merge the PR.

**Pull request merged**